net/bluetooth/bluetooth_conn.c

```diff
@@ -42,10 +42,19 @@
 #include <nuttx/net/bluetooth.h>
 
 #include "devif/devif.h"
+#include "utils/utils.h"
 #include "bluetooth/bluetooth.h"
 
 #ifdef CONFIG_NET_BLUETOOTH
 
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifndef CONFIG_NET_BLUETOOTH_MAX_CONNS
+#  define CONFIG_NET_BLUETOOTH_MAX_CONNS 0
```

Comment:
@wengzhe why to use 0 as max connections? Please for logic reasoning use 1 and later in your logic operation decrease 1

Reply:
It's because the default value of max_conns is 0 (https://github.com/apache/nuttx/blob/master/net/bluetooth/Kconfig#L57) and I found it means no limit because everywhere we use it to limit connections is checking `#if max_conns > 0`. In this PR I set the `sem` to `INT16_MAX` in `net_bufpool_init` to get similar behavior.